crates/rooch-types/src/address.rs

```diff
@@ -810,19 +811,29 @@ impl TryFrom<MultiChainAddress> for BitcoinAddress {
     }
 }
 
+// TODO: rename NostrAddress to NostrPublicKey? https://github.com/nostr-protocol/nips/blob/master/19.md
 /// Nostr address type
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct NostrAddress(pub XOnlyPublicKey);
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct NostrAddress {
+    pub hex: XOnlyPublicKey,
+    pub bech32: String,
```

Comment:
We can just keep the XOnlyPublicKey field in NostrAddress, and output bech32 in the display function.

Reply:
Resolved but there needs to deal with the test function to match hex and bech32 addresses.